Allow building GNU Privacy Guard's dependencies

dscho opened the pull request:
Since MSYS2 stopped servicing their i686 flavor, Git for Windows has to take care of building all kinds of things that we formerly got "for free".

To build `gnupg`, we need an updated `libgcrypt`. For that, we need a newer `libgpg-error` (at least we _thought_ that we did). And to build `libgpg-error`, we need to switch from the `pkg-config` to the `pkgconf` package.

This PR allows us to do all that.

dscho commented:
The test still fails 🙄 

Let's do https://github.com/git-for-windows/MSYS2-packages/pull/44 first, and then graduate to this here PR.

rimrul commented on .ci/ci-build.sh:
```diff
@@ -26,6 +26,7 @@ list_commits  || failure 'Could not detect added commits'
 list_packages || failure 'Could not detect changed files'
 message 'Processing changes' "${commits[@]}"
 test -z "${packages}" && success 'No changes in package recipes'
+define_build_order || failure 'Could not determine build order'
```
So far so good. What changed about the hang?

dscho replied:
The fact that I found out _why_ it "hung": I failed to realize that `actions/checkout@v2` performs a shallow clone by default, which made `git log upstream/main..` look as if _every single package_ was updated, and hence `define_build_order` (which is quadratic in the number of packages, if I remember correctly) worked accordingly hard... 😁

rimrul replied:
That explains the unexpectedly long time to run.

rimrul commented on libgcrypt/fix-broken-mlock-detection.patch:
```diff
@@ -0,0 +1,58 @@
+Subject: [PATCH] Fix broken mlock detection
+
+We need to be careful when casting a pointer to a `long int`: the
+highest bit mit be set, in which case the result is a negative number.
```
Nit: s/mit be/might be/

dscho replied:
Good point.

rimrul commented on libgpg-error/libgpg-error-1.16-msys2.patch:
```diff
@@ -29,8 +29,8 @@ diff -Naur libgpg-error-1.16/build-aux/config.guess libgpg-error-1.16-msys2/buil
 +    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
 +	echo x86_64-unknown-msys
 +	exit ;;
```
Do we also need a i686 version of this?

dscho replied:
Maybe? I added it, just in case.

dscho commented:
I hadn't realized that the build definition needed to be adjusted to expect certain GPG signatures. New run: https://dev.azure.com/Git-for-Windows/git/_build/results?buildId=79104&view=results

dscho commented:
_Of course_ there was another problem. I had to change the task that calls `rebaseall` (for some reason I don't remember, the `msys2-runtime` build failed if I rebased, so I guarded that task behind an include list; I changed that to an exclude list now). New run: https://dev.azure.com/Git-for-Windows/git/_build/results?buildId=79106&view=results

dscho commented:
Building `libgcrypt` now: https://dev.azure.com/Git-for-Windows/git/_build/results?buildId=79111&view=results

dscho commented:
> 
> 
> Building `libgcrypt` now: https://dev.azure.com/Git-for-Windows/git/_build/results?buildId=79111&view=results

It worked. With that done, I'll head over to https://github.com/git-for-windows/git/issues/3267.
